### PR TITLE
FullVM ubuntu 22 and 24.04

### PR DESCRIPTION
**Pull request by PeterNashaat**
- Updated flists directories with 2 sub directories Fullvm and microvm
- Added automation script which automates the setup, configuration, archiving, and uploading of an Ubuntu system ready for use as an FLIST. 
- Testing Full vms  for now is done through Terrafrom
- Added readme.md file for Automated, Manual and Testing process

### CODE REVIEW

**Review by coesensbert — APPROVED** [review]

File: tfgrid3/ubuntu24.04/fullvm/ubuntu24-fullvm-flist-builder.sh [hunk 1]
```diff
@@ -68,4 +68,4 @@ if [ -z "$API_KEY" ]; then
 	        exit 2
 fi
 
-curl -X POST -H "Authorization: Bearer $API_KEY" -F "file=@ubuntu-jammy.tar.gz" https://hub.grid.tf/api/flist/me/upload
\ No newline at end of file
+curl -X POST -H "Authorization: Bearer $API_KEY" -F "file=@ubuntu-noble.tar.gz" https://hub.grid.tf/api/flist/me/upload
```
Comment (hossnys) [inline tfgrid3/ubuntu24.04/fullvm/ubuntu24-fullvm-flist-builder.sh]: could you please use the version number in the name of the flist better than using image codename to be more clear for users when adding it to hub then dashboard

File: tfgrid3/ubuntu24.04/fullvm/ubuntu24-fullvm-flist-builder.sh [hunk 2]
```diff
@@ -55,11 +55,11 @@ if ! command -v extract-vmlinux &>/dev/null; then
...
 
 # Kernel Extraction
-extract-vmlinux ubuntu-jammy/boot/vmlinuz | tee ubuntu-jammy/boot/vmlinuz-5.15.0-25-generic.elf > /dev/null
-mv ubuntu-jammy/boot/vmlinuz-5.15.0-25-generic.elf ubuntu-jammy/boot/vmlinuz-5.15.0-25-generic
+extract-vmlinux ubuntu-noble/boot/vmlinuz | tee ubuntu-noble/boot/vmlinuz-6.8.0-31-generic.elf > /dev/null
+mv ubuntu-noble/boot/vmlinuz-6.8.0-31-generic.elf ubuntu-noble/boot/vmlinuz-6.8.0-31-generic
 
 # Create a compressed archive of the configured system for uploading to hub.
-tar -czf ubuntu-jammy.tar.gz -C ubuntu-jammy .
+tar -czf ubuntu-noble.tar.gz -C ubuntu-noble .
```
Comment (hossnys) [inline tfgrid3/ubuntu24.04/fullvm/ubuntu24-fullvm-flist-builder.sh]: could you please use the version number in the name of the flist better than using image codename to be more clear for users when adding it to hub then dashboard

File: tfgrid3/ubuntu22.04/microvm/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Development Guide for Ubuntu:22.04 image
+## This guide will take you through steps for pulling & running ubuntu:22.04 image
+
+### Pull the image
+```bash
+docker pull threefolddev/ubuntu:22.04
```
Comment (hossnys) [inline tfgrid3/ubuntu22.04/microvm/README.md]: images are not pushed there on threefolddev docker hub account 
Screenshot from 2024-05-16 11-52-35

- so it's needed to push latest image there to keep this doc valid .

File: tfgrid3/ubuntu22.04/fullvm/README.md [hunk 4]
```diff
@@ -69,4 +69,6 @@ tar -czf ubuntu-jammy.tar.gz -C ubuntu-jammy .
 clsecret="$API_KEY"
 curl -X Post -H "Authorization: Bearer ${clsecret}" -F "file=@ubuntu-jammy.tar.gz"  https://hub.grid.tf/api/flist/me/upload
```
Comment (hossnys) [inline tfgrid3/ubuntu22.04/fullvm/README.md:70]: could you please use the version number in the name of the flist better than using image codename to be more clear for users when adding it to hub then dashboard

File: tfgrid3/ubuntu24.04/microvm/zinit/ssh-init.yaml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+exec: bash /start.sh
```
Comment (hossnys) [inline tfgrid3/ubuntu24.04/microvm/zinit/ssh-init.yaml]: better to use ```exec: start.sh ``` directly to just use one process which is exec , not two which is zinit and bash

File: tfgrid3/ubuntu24.04/microvm/README.md [hunk 6]
```diff
@@ -1,17 +1,17 @@
-# Development Guide for Ubuntu:22.04 image
-## This guide will take you through steps for pulling & running ubuntu:22.04 image
+# Development Guide for Ubuntu:24.04 Noble image
+## This guide will take you through steps for pulling & running ubuntu:24.04 image
 
 ### Pull the image
 ```bash
-docker pull threefolddev/ubuntu:22.04
+docker pull threefolddev/ubuntu:24.04
```
Comment (hossnys) [inline tfgrid3/ubuntu24.04/microvm/README.md:6]: docker image is not pushed to threefolddev docker hub account , so this command will not work .
Screenshot from 2024-05-16 12-19-02

Comment (PeterNashaat) [reply tfgrid3/ubuntu24.04/microvm/README.md:6]: pushed docker images

File: tfgrid3/ubuntu24.04/microvm/zinit/ssh-init.yaml (hunk 5 above)
Comment (PeterNashaat) [reply tfgrid3/ubuntu24.04/microvm/zinit/ssh-init.yaml]: - updated

File: tfgrid3/ubuntu22.04/fullvm/README.md (hunk 4 above)
Comment (PeterNashaat) [reply tfgrid3/ubuntu22.04/fullvm/README.md:70]: pushed docker images

File: tfgrid3/ubuntu22.04/microvm/README.md (hunk 3 above)
Comment (PeterNashaat) [reply tfgrid3/ubuntu22.04/microvm/README.md]: pushed docker images

File: tfgrid3/ubuntu24.04/fullvm/ubuntu24-fullvm-flist-builder.sh (hunk 2 above)
Comment (PeterNashaat) [reply tfgrid3/ubuntu24.04/fullvm/ubuntu24-fullvm-flist-builder.sh]: - updated

File: tfgrid3/ubuntu24.04/fullvm/ubuntu24-fullvm-flist-builder.sh (hunk 1 above)
Comment (PeterNashaat) [reply tfgrid3/ubuntu24.04/fullvm/ubuntu24-fullvm-flist-builder.sh]: updated

**Review by hossnys — APPROVED** [review]

**Pull request merged**